python/cog/files.py

```diff
@@ -54,15 +54,22 @@ def put_file_to_signed_endpoint(
     resp = client.put(
         ensure_trailing_slash(endpoint) + filename,
         fh,  # type: ignore
-        headers={"Content-type": content_type},
+        headers={
+            "Content-Type": content_type,
+            "X-Prediction-ID": prediction_id,
+        },
```

Comment:
We probably don't want to set `X-Prediction-ID` if `prediction_id` is `None`.

```suggestion
        headers={"Content-Type": content_type}

        if prediction_id is not None:
            headers["X-Prediction-ID"] = prediction_id
```